onnxruntime/core/session/inference_session.cc

```diff
@@ -1748,23 +1749,21 @@ common::Status InferenceSession::Initialize() {
                        [](char ch) { return std::tolower(ch); });
         bool dml_graph_serialization_enabled = dml_graph_serialization_enabled_config_val == "true";
 
-        if (dml_graph_fusion_enabled) {
+        if (static_cast<const Dml::ExecutionProvider*>(dmlExecutionProvider)->IsGraphCaptureEnabled()) {
+          std::unique_ptr<onnxruntime::GraphTransformer> dmlRuntimeGraphFusionTransformer = std::make_unique<Dml::DmlRuntimeGraphFusionTransformer>("DmlRuntimeGraphFusionTransformer",
+                                                                                                                                                    dmlExecutionProvider);
+          if (dmlRuntimeGraphFusionTransformer == nullptr) {
+            return Status(common::ONNXRUNTIME, common::FAIL, "DmlRuntimeGraphFusionTransformer is nullptr");
+          }
+          ORT_RETURN_IF_ERROR_SESSIONID_(graph_transformer_mgr_.Register(std::move(dmlRuntimeGraphFusionTransformer), onnxruntime::TransformerLevel::Level3));
+        } else if (dml_graph_fusion_enabled) {
```

Comment:
Does it mean only one of them can be enabled at a time?

Reply:
Yeah, otherwise it partitions them into different partitions and perf ends up being worse.